### PR TITLE
Miscellaneous doc fixes

### PR DESCRIPTION
**Pull request by abidlabs**
* Improve documentation for `gr.Audio` component, particularly the shape of the audio data array. Closes: https://github.com/gradio-app/gradio/issues/6578
* Clarify that `every` only applies to backend functions, not `js` functions. Closes: https://github.com/gradio-app/gradio/issues/5365
* Fix the example for Interface state and improve the docs there generally. Closes: https://github.com/gradio-app/gradio/issues/4665

### CODE REVIEW

File: gradio/components/audio.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ class Audio(
 ):
     """
     Creates an audio component that can be used to upload/record audio (as an input) or display audio (as an output).
-    Preprocessing: passes the uploaded audio as a {Tuple(int, numpy.array)} corresponding to (sample rate in Hz, audio data as a 16-bit int array whose values range from -32768 to 32767), or as a {str} filepath, depending on `type`.
+    Preprocessing: depending on `type`, passes the uploaded audio as {str} filepath or a {Tuple(int, numpy.array)} corresponding to (sample rate in Hz, audio data). If the latter, the audio data is a 16-bit int array whose values range from -32768 to 32767 and shape of the audio data array is (samples,) for mono audio or (samples, channels) for multi-channel audio.
```
Comment (aliabd) [inline gradio/components/audio.py:52]:
```suggestion
    Preprocessing: depending on `type`, passes the uploaded audio as {str} filepath or a {Tuple(int, numpy.array)} corresponding to (sample rate in Hz, audio data). If the latter, the audio data is a 16-bit int array whose values range from -32768 to 32767 and shape of the audio data array is {(samples,)} for mono audio or {(samples, channels)} for multi-channel audio.
```

**Review by aliabd — APPROVED** [review]
Thanks for the fixes @abidlabs They all lgtm! Consider adding the new `interface_state` demo as a demo in the State class docstring. so it shows up in the docs too.

File: gradio/components/audio.py (hunk 1 above)
Comment (abidlabs) [reply gradio/components/audio.py:52]: Why the curly brackets?

Comment (aliabd) [reply gradio/components/audio.py:52]: Adds formatting that makes it easier to parse the text. Feel free to ignore

Screen Shot 2024-01-24 at 11 00 35 AM

Screen Shot 2024-01-24 at 11 03 35 AM

Comment (abidlabs) [reply gradio/components/audio.py:52]: I thought we only used that to designate Python types

**Comment by abidlabs** [comment]
Thanks @aliabd! I’ll add the to docs

**Pull request merged**